nomt/src/beatree/ops/update/leaf_stage.rs

```diff
@@ -308,7 +309,9 @@ fn try_answer_left_neighbor(
 
     let consumed_whole_range = next_separator == worker_params.range.high;
 
-    if !found_next_node && !consumed_whole_range {
+    // wait to respond if no next node was found or if the entire range was consumed but
+    // the worker didn't finish to interact with the right neighbor
+    if !found_next_node && !(consumed_whole_range && has_finished_with_right_neighbor) {
```

Comment:
To me this looks more like `consumed_whole_range` doesn't do what I originally meant it to do. We can probably just remove the notion of `consumed_whole_range` altogether and have `has_finished_with_right_neighbor` do all the heavy lifting, re: https://github.com/thrumdev/nomt/issues/425

The intention is to link up our left and right neighbor once we've finished our work. 

as you've correctly found, there are issues with the current `consumed_whole_range` property.
 1. if we're still interacting with the right neighbor to e.g. complete merges, we can answer too early and cause issues. 
 2. in #425 you note that workers may never consume their whole range (and likely don't)
 
So I suggest a refactor to remove the notion of `consumed_whole_range` and replace it with `has_finished_with_right_neighbor`.

Reply:
`consumed_whole_range` has been removed in #426